rj_gameplay/rj_gameplay/tactic/goalie_tactic.py

```diff
@@ -113,61 +190,80 @@ def get_requests(self, world_state: rc.WorldState,
         # max out of box to cap for goalie
         MAX_OOB = RobotConstants.RADIUS
 
+        ball_speed = np.linalg.norm(world_state.ball.vel)
+        ball_pos = world_state.ball.pos
+        ball_dist = np.linalg.norm(world_state.field.our_goal_loc -
+                                   ball_pos)
+        goal_pos = world_state.field.our_goal_loc
+        towards_goal = goal_pos - ball_pos
+
         role_requests = {}
-        if world_state and world_state.ball.visible:
-            ball_speed = np.linalg.norm(world_state.ball.vel)
-            ball_pos = world_state.ball.pos
-            ball_dist = np.linalg.norm(world_state.field.our_goal_loc - ball_pos)
-            goal_pos = world_state.field.our_goal_loc
-            towards_goal = goal_pos - ball_pos
+        best_utility = utility_functions()
+
+        if best_utility is not None:
 
+            # what is this?
```

Comment:
oh dear someone else found it

This is a hack I put in during comp2021 to make sure our goalie didn't try to intercept the ball in stop/penalty situations. (All robots are supposed to stop moving when a foul is called or when a penalty kick is being set up.)

Reply:
```suggestion
            # TODO (1724) remove this from the code
```
https://github.com/RoboJackets/robocup-software/issues/1724